android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -107,8 +106,23 @@ private JSONObject jsonFromErrorMessageString(String errorMessage) throws JSONEx
    @ReactMethod 
    public void init(String appId) {
       Activity activity = getCurrentActivity();
-      if (activity == null || oneSignalInitDone) {
-         Log.e("onesignal", "Unable to initialize the OneSignal SDK because activity is null " + (activity == null) + " or oneSignalInitDone" + oneSignalInitDone);
+      
+      if (activity == null) {
```

Comment:
Not sure what values are on `mReactApplicationContext` but can we get an Android `Context` from this if `getCurrentActivity()` gives us `null`?

Reply:
@jkasten2 `getCurrentActivity()` is what is now used in RN to get the current activity. The old way of getting the activity (from the ReactApplicationContext object) is deprecated

https://github.com/facebook/react-native/releases/tag/v0.29.0

Reply:
I believe `mReactApplicationContext.getApplicationContext()` should always be non-null.

Reply:
@jkasten2 `mReactApplicationContext.getApplicationContext()` behaves the same way. 

It seems that as of 2016, React-Native no longer guarantees this method will return a non null activity at the start of the app. So I think I'll just change it to get the activity from `mReactApplicationContext` and still check for null, and if it's null, re-check in 50ms. It appears this resolves a pretty common issue in RN applications.